src/btree/bt_page.c

```diff
@@ -797,22 +797,20 @@ __inmem_row_int(WT_SESSION_IMPL *session, WT_PAGE *page, size_t *sizep)
             break;
         case WT_CELL_ADDR_DEL:
             /*
-             * A cell may reference a deleted leaf page: if a leaf page was deleted without being
-             * read (fast truncate), and the deletion committed, but older transactions in the
-             * system required the previous version of the page to remain available, a special
-             * deleted-address type cell is written. We'll see that cell on a page if we read from a
-             * checkpoint including a deleted cell or if we crash/recover and start off from such a
-             * checkpoint (absent running recovery, a version of the page without the deleted cell
-             * would eventually have been written). If we crash and recover to a page with a
-             * deleted-address cell, we want to discard the page from the backing store (it was
-             * never discarded), and, of course, by definition no earlier transaction will ever need
-             * it.
-             *
-             * Re-create the state of a deleted page.
+             * If a page was deleted without being read (fast truncate), and the delete committed,
+             * but older transactions in the system required the previous version of the page to
+             * remain available or the delete can still be rolled back by RTS, a deleted-address
+             * type cell is written. We'll see that cell on a page if we read from a checkpoint
+             * including a deleted cell or if we crash/recover and start off from such a checkpoint.
+             * Recreate the fast-delete state for the page.
              */
-            ref->addr = unpack.cell;
+            if (F_ISSET(page->dsk, WT_PAGE_FT_UPDATE)) {
+                WT_ERR(__wt_calloc_one(session, &ref->ft_info.del));
+                *ref->ft_info.del = unpack.page_del;
+                ref->ft_info.del->previous_state = WT_REF_DISK;
+                ref->ft_info.del->committed = 1;
```

Comment:
Can we change committed to a bool instead of uint8_t? I know it may impact backward compatibility.

Reply:
I don't want to touch the `WT_PAGE_DELETED` structure in this change, but we could create another ticket if you think it's worth it.

Reply:
@quchenhao, I ended up changing WT_PAGE_DELETED, so I fixed this in a1425d7.